src/gui/TimeLineWidget.cpp

```diff
@@ -304,60 +307,103 @@ void TimeLineWidget::paintEvent( QPaintEvent * )
-		{
-			m_moveXOff = event->x() - m_xOffset;
-		}
+	// TODO: Read these from a config
+	auto leftCtrlAction = DragLoop;
+	auto rightCtrlAction = MoveLoopClosest;
+	auto leftShiftAction = MoveLoopBegin;
+	auto rightShiftAction = MoveLoopEnd;
+	auto button = event->button();
+	auto mods = event->modifiers();
+	
+	// Unmodified LMB is reserved for playhead, unmodified RMB for context menu
+	// Shift or Ctrl modified press behavior is bound by the user
+	// Shift + Ctrl modifier is reserved for fine adjustment of Shift actions
+	// TODO: Let MMB be bound
+	
+	if (event->x() < m_xOffset) { return; }
+	
+	// Choose an action based on input event and user's bindings
+	else if (button == Qt::LeftButton) {
+		if (mods & Qt::ShiftModifier){ m_action = leftShiftAction; }
+		else if (mods == Qt::ControlModifier){ m_action = leftCtrlAction; }
+		// Handles moving playhead
 		else
 		{
-			m_moveXOff = s_posMarkerPixmap->width() / 2;
+			m_action = MovePositionMarker;
+			if (event->x() - m_xOffset < s_posMarkerPixmap->width())
+			{
+				m_moveXOff = event->x() - m_xOffset;
+			}
+			else { m_moveXOff = s_posMarkerPixmap->width() / 2; }
 		}
 	}
-	else if( event->button() == Qt::LeftButton  && (event->modifiers() & Qt::ShiftModifier) )
+	else if (button == Qt::RightButton) {
+		if (mods & Qt::ShiftModifier){ m_action = rightShiftAction; }
+		else if (mods == Qt::ControlModifier){ m_action = rightCtrlAction; }
+		else { contextMenuEvent(nullptr); }
+	}
+	
+	// Calculate the clicked position as a TimePos, several actions need this
+	m_moveXOff = s_posMarkerPixmap->width() / 2;
+	const auto ticksPerPixel = TimePos::ticksPerBar() / m_ppb;
+	const TimePos t = m_begin + static_cast<int>(qMax(event->x() - m_xOffset - m_moveXOff, 0) * ticksPerPixel);
+	
+	// Translate MoveLoopClosest into left or right based on distance
+	if (m_action == MoveLoopClosest)
 	{
-		m_action = SelectSongTCO;
-		m_initalXSelect = event->x();
+		const TimePos loopMid = (m_loopPos[0] + m_loopPos[1]) / 2;
+		if (t < loopMid) { m_action = MoveLoopBegin; }
+		else { m_action = MoveLoopEnd; }
 	}
-	else if( event->button() == Qt::RightButton )
+	// If we're moving loop markers or dragging loop markers, update them
+	if (m_action == MoveLoopBegin){ m_loopPos[0] = t; }
+	else if (m_action == MoveLoopEnd){ m_loopPos[1] = t; }
+	else if (m_action == DragLoop)
 	{
-		m_moveXOff = s_posMarkerPixmap->width() / 2;
-		const TimePos t = m_begin + static_cast<int>( qMax( event->x() - m_xOffset - m_moveXOff, 0 ) * TimePos::ticksPerBar() / m_ppb );
-		const TimePos loopMid = ( m_loopPos[0] + m_loopPos[1] ) / 2;
-
-		if( t < loopMid )
-		{
-			m_action = MoveLoopBegin;
-		}
-		else if( t > loopMid )
-		{
-			m_action = MoveLoopEnd;
-		}
-
-		if( m_loopPos[0] > m_loopPos[1]  )
-		{
-			qSwap( m_loopPos[0], m_loopPos[1] );
-		}
-
-		m_loopPos[( m_action == MoveLoopBegin ) ? 0 : 1] = t;
+		const bool quant = !(mods == (Qt::ControlModifier | Qt::ShiftModifier));
```

Comment:
That wasn't the intended behavior, but yeah, that's how it turned out. That seems fine to me so I'll update the tooltip instead of trying to prevent it.